### ISSUE TITLE
Bug for the implementation of knowledge distillation?

### ISSUE THREAD

**Issue by TingAnChien** [mit-han-lab/once-for-all#5]
Thanks for sharing your code!
I'm wondering if this is a bug for the implementation of knowledge distillation.
Since the `cross_entropy_loss_with_soft_target` already use `nn.LogSoftmax`,
https://github.com/mit-han-lab/once-for-all/blob/0807662ec0354c98c808496d540518badb6b9a9b/imagenet_codebase/utils/pytorch_utils.py#L37

Does it need to apply `softmax` on `soft_logits` here again? Thanks!
https://github.com/mit-han-lab/once-for-all/blob/0807662ec0354c98c808496d540518badb6b9a9b/elastic_nn/training/progressive_shrinking.py#L115

**Comment by TingAnChien** [comment]
Oh, I thought I was wrong. Please ignore this issue, thanks.
